### PR TITLE
Support container info from v1.1.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
This uses the `ContainerInfo` in a new USI `TaskBuilder` to the Mesos
container configuration to a `TaskInfo` for a new Mesos task.

The following configuration options were dropped:
* `executorCpus` and `executorMem`
* `remoteFSRoot`
* `jvmArgs`
* `defaultSlave`
* `slaveAttributes`
* In `ContainerInfo`
  * `parameters`
  * `networking`, defaults to `HOST`
  * `portMappings`
  * `networkInfos`
  * `useCustomDockerCommandShell` and `customDockerCommandShell`
  * `dockerImageCustomizable`

They can be added back in if required.

JIRA issues: DCOS_OSS-5587

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/mesos/MesosAgentSpecTemplate.java [hunk 1]
```diff
@@ -178,7 +181,213 @@ public String getJnlpArgs() {
...
+    private final List<PortMapping> portMappings;
+    private final List<NetworkInfo> networkInfos;
+    private final boolean useCustomDockerCommandShell;
+    private final String customDockerCommandShell;
+    private final boolean dockerPrivilegedMode;
+    private final boolean dockerForcePullImage;
+    private final boolean dockerImageCustomizable;
+    private boolean isDind;
+
+    public ContainerInfo(
```
Comment (jeschkies) [inline src/main/java/org/jenkinsci/plugins/mesos/MesosAgentSpecTemplate.java:212]: What should we use? What should be deprecated?

File: src/main/java/org/jenkinsci/plugins/mesos/api/RunTemplateFactory.java [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+    }
+
+    /**
+     * This is the original v1.1 JenkinsScheduler.getContainerInfoBuilder.
+     *
+     * @param offer The Mesos offer.
+     * @param agentName ???
+     * @param taskBuilder The Mesos task info builder.
+     */
+    private void getContainerInfoBuilder(
```
Comment (jeschkies) [inline src/main/java/org/jenkinsci/plugins/mesos/api/RunTemplateFactory.java]: This method is almost a verbatim copy of [JenkinsScheduler.getContainerInfoBuilder](https://github.com/jenkinsci/mesos-plugin/blob/1.x/src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java#L879). I think it should be cleaned up.

File: src/main/java/org/jenkinsci/plugins/mesos/api/RunTemplateFactory.java [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+        Map<TaskName, Seq<Resource>> peerTaskResources) {
+      this.simpleTaskInfoBuilder.buildTask(builder, matchedOffer, taskResources, peerTaskResources);
+      this.getContainerInfoBuilder(matchedOffer, "what name?", builder);
+    }
+
+    /**
+     * This is the original v1.1 JenkinsScheduler.getContainerInfoBuilder.
+     *
+     * @param offer The Mesos offer.
+     * @param agentName ???
```
Comment (jeschkies) [inline src/main/java/org/jenkinsci/plugins/mesos/api/RunTemplateFactory.java]: @kaiwalyajoshi and @kvish what are the semantics of `agentName`/`slaveName`?

File: src/main/java/org/jenkinsci/plugins/mesos/api/RunTemplateFactory.java [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+   * ContainerInfoTaskInfoBuilder}.
+   *
+   * @param requirements The resource requirements for a Jenkins agent.
+   * @param shellCommand The shell command built by {@link LaunchCommandBuilder}.
+   * @param role The Mesos role the Jenkins agent will assume.
+   * @param fetchUris Artifacts that are fetched, eg the Jenkins agent.jar.
+   * @param containerInfo Optional information for a Docker or Mesos container.
+   * @return the new USI run template.
+   */
+  static RunTemplate newRunTemplate(
```
Comment (jeschkies) [inline src/main/java/org/jenkinsci/plugins/mesos/api/RunTemplateFactory.java:50]: @timcharper, is this pattern fine? I simply wrap `SimpleTaskInfoBuilder`.

File: src/main/java/org/jenkinsci/plugins/mesos/MesosAgentSpecTemplate.java [hunk 5]
```diff
@@ -178,7 +181,213 @@ public String getJnlpArgs() {
     return jnlpArgs;
   }
 
-  public String getContainerImage() {
-    return this.containerImage;
+  public ContainerInfo getContainerInfo() {
+    return this.containerInfo;
+  }
+
+  public static class ContainerInfo {
```
Comment (jeschkies) [inline src/main/java/org/jenkinsci/plugins/mesos/MesosAgentSpecTemplate.java]: This is moved from `MesosSlaveInfo`.

File: src/main/java/org/jenkinsci/plugins/mesos/MesosAgentSpecTemplate.java (hunk 1 above)
Comment (kaiwalyajoshi) [reply src/main/java/org/jenkinsci/plugins/mesos/MesosAgentSpecTemplate.java:212]: IMHO, I found `ContainerInfo` to be pretty confusing between the Jenkins Scheduler context and Mesos's ContainerInfo. 

I propose renaming this to something more friendly that will convey the same requirements to USI.

**Pull request merged**